Update spikeinterface to the last version

h-mayorquin opened the pull request:
Update the comparison function to take into account the latest changes in the NWBRecordingExtractor in Spikeinterface.

CodyCBakerPhD commented:
Looks like this cleans up all the SI related issues

Can you add a CHANGELOG describing what to expect for the new behavior? I get they're all casted as strings now, but what was the previous behavior? (I thought they would always be strings in NWB file in the end)

h-mayorquin commented on src/neuroconv/tools/testing/data_interface_mixins.py:
```diff
@@ -314,6 +323,20 @@ def check_read_nwb(self, nwbfile_path: str):
                 if recording.has_scaled_traces() and self.nwb_recording.has_scaled_traces():
                     check_recordings_equal(RX1=recording, RX2=self.nwb_recording, return_scaled=True)
 
+            ## Compare channel groups
```
@CodyCBakerPhD and added here to

h-mayorquin commented:
> Looks like this cleans up all the SI related issues
> 
> Can you add a CHANGELOG describing what to expect for the new behavior? I get they're all casted as strings now, but what was the previous behavior? (I thought they would always be strings in NWB file in the end)

Added. `NWBRecordingExtractor` always is using `group_name` to set the group property. Previously, we were comparing `group` (in the original recording) to `group` (in the `NWBRecordingExtractor`). I added logic to compare `group name` to group when this exists.

Basically,  I am reflecitng the writing logic here:
https://github.com/catalystneuro/neuroconv/blob/374ad9d89f15a1be1407134975e85e3ecb10b8dc/src/neuroconv/tools/spikeinterface/spikeinterface.py#L279-L287

On the comparison testing here:
https://github.com/catalystneuro/neuroconv/blob/374ad9d89f15a1be1407134975e85e3ecb10b8dc/src/neuroconv/tools/testing/data_interface_mixins.py#L325-L335
